File: frontend/src/services/departmentService.js
import axiosClient from "../api/axiosClient";

const departmentService = {

    getDepartments: async () => {

        const response = await axiosClient.get("/departments");

        return response.data;

    },

    getActiveDepartments: async () => {

        const response = await axiosClient.get("/departments/active");

        return response.data;

    },

    saveDepartment: async (department) => {

        const response = await axiosClient.post(
            "/departments",
            department
        );

        return response.data;

    }

};

export default departmentService;

File: backend/src/main/java/com/cdac/itregister/service/DepartmentService.java
package com.cdac.itregister.service;

import com.cdac.itregister.dto.ApiResponse;
import com.cdac.itregister.entity.Department;

public interface DepartmentService {

    ApiResponse<?> saveDepartment(Department department);

    ApiResponse<?> getDepartments();

    ApiResponse<?> getActiveDepartments();
    ApiResponse<?> getAllDepartments();

    ApiResponse<?> updateDepartmentStatus(
            Long id,
            String status
    );

}